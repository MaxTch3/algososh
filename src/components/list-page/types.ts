export class ListNode<T> {
   value: T;
   next: ListNode<T> | null;

   constructor(value: T) {
      this.value = value;
      this.next = null
   }
}

export interface IList<T> {
   push: (value: T) => void;
   unshift: (value: T) => void;
   insertAt: (value: T, index: number) => void;
   deleteAt: (index: number) => void
}
