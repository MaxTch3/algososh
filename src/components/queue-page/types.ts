import { ElementStates } from '../../types/element-states';

export interface IQueue<T> {
   enqueue: (item: T) => void;
   dequeue: () => void;
   getQueue: () => Array<T | undefined>
}

export interface IQueueItem {
   state?: ElementStates;
   value?: string;
}