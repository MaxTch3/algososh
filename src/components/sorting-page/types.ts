import { ElementStates } from '../../types/element-states';

export interface IColumn {
   number: number;
   state: ElementStates
};

export enum AlgorithmMethod { SelectionSort, BubbleSort };
