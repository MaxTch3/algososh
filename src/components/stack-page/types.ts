import { ElementStates } from '../../types/element-states';

export interface IStack<T> {
   push: (item: T) => void;
   pop: () => void;
   reset: () => void
}

export interface IStackItem {
   state?: ElementStates;
   head?: string;
   value: string
}