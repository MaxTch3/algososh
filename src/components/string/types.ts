import { ElementStates } from '../../types/element-states';

export interface IItem{
   state: ElementStates,
   item: string
}