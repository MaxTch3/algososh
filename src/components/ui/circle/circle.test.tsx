import { render } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe('Тест компонента Circle', () => {
   it('Отрисовка без символов', () => {
      const circle = render(<Circle />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с символами', () => {
      const circle = render(<Circle letter='К' />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с head', () => {
      const circle = render(<Circle head='Р' />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с react-элементом в head', () => {
      const circle = render(
         <Circle head={<Circle isSmall={true} letter='У' />} />
      ).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с tail', () => {
      const circle = render(<Circle tail='Г' />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с react-элементом в tail', () => {
      const circle = render(
         <Circle tail={<Circle isSmall={true} letter='И' />} />
      ).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с index', () => {
      const circle = render(<Circle index={0} />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка с пропом isSmall === true', () => {
      const circle = render(<Circle isSmall={true} />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка в состоянии default', () => {
      const circle = render(<Circle state={ElementStates.Default} />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка в состоянии changing', () => {
      const circle = render(<Circle state={ElementStates.Changing} />).container;
      expect(circle).toMatchSnapshot();
   });

   it('Отрисовка в состоянии modified', () => {
      const circle = render(<Circle state={ElementStates.Modified} />).container;
      expect(circle).toMatchSnapshot();
   });
})