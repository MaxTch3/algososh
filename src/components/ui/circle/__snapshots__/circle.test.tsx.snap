// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Circle Отрисовка без символов 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка в состоянии changing 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   changing"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка в состоянии default 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка в состоянии modified 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   modified"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
      test-id="head"
    >
      Р
    </div>
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      0
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с react-элементом в head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          test-id="head"
        />
        <div
          class="circle  small default"
          test-id="smallCircle"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            У
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          test-id="tail"
        />
      </div>
    </div>
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с react-элементом в tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          test-id="head"
        />
        <div
          class="circle  small default"
          test-id="smallCircle"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            И
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          test-id="tail"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
      test-id="tail"
    >
      Г
    </div>
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с пропом isSmall === true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle  small default"
      test-id="smallCircle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;

exports[`Тест компонента Circle Отрисовка с символами 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      test-id="head"
    />
    <div
      class="circle   default"
      test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        К
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      test-id="tail"
    />
  </div>
</div>
`;
